core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -152,6 +154,14 @@ class LogManager(logDirs: Array[File],
       Exit.halt(1)
     }
 
+    // If IBP < 1.0, controller will send LeaderAndIsrRequest V0 which does not include isNew field.
```

Comment:
We could also just add offline log dirs to logDirFailureChannel and let LogDirFailureHandler handle it since it already has the logic for dealing with IBP. Would that be better?

Reply:
Thanks for the comment. It seems that the current approach is better. LogDirFailureHandler is only started after ReplicaManager.start() is called. But LogManager.createAndValidateLogDirs() is called when the LogManager is instantiated, which is before ReplicaManager.start() is called. I think we probably want to halt the broker immediately when it has enough information to do so, rather than waiting for ReplicaManager to start.

Reply:
Well, the approach of going through logDirFailureChannel requires less code, which is an advantage. It does delay the shutting down of the broker a bit, but is there any harm in doing that?

Reply:
There is no harm with either approach. I just feel that the extra lines of code is less of any issue than shutting down broker a bit. Sure. I have updated the patch as suggested.

The newly added test failed. I am not sure why.. I will fix this test later today.

Reply:
I have rebased patch onto trunk and fixed test failure.